packages/core/cli.js

```diff
@@ -41,40 +41,21 @@ listeners.forEach(listener => process.removeListener("warning", listener));
 
 const inputStrings = process.argv.slice(2);
 
+const { handleHelpInput } = require("./lib/cliHelp");
 const {
   getCommand,
   prepareOptions,
   runCommand,
   displayGeneralHelp
 } = require("./lib/command-utils");
 
-//User only enter truffle with no commands, let's show them what's available.
-if (inputStrings.length === 0) {
+// handle cases where input indicates the user wants to access Truffle's help
+const { displayHelp } = handleHelpInput({ inputStrings });
```

Comment:
`handleHelpOutput` is too innocent a name for what the function does. It returns a boolean and mutates `inputString`.  What do you think about different name, and return type, so there's no side effect? (`line 42: const inputStrings = ...` would have to be deleted, though.)

```
const { displayName, inputStrings } = parseArguments(proces.argv.slice(2));
```

Reply:
I don't know, I kind of feel like this is not worth agonizing over. `handleHelpInput` doesn't bother me, maybe a comment about what it does might suffice?

Reply:
I'd rather refactor this utility/helper function to be a pure function because it's now a module, and it adds another bit of technical debt. If others think a comment would suffice, I'll be ok with that.  @fainashalts @haltman-at, @gnidan, thoughts?

Reply:
I guess what I'm really saying is that in general I think our time would be better spent fixing technical debt on much more important and complicated features rather than agonizing over help display. However, I'm not going to die on this hill so will rewrite it to be pure.